### DISCUSSION TITLE
Project Type Recategorization

### DISCUSSION THREAD

**Discussion by qixils** [modrinth/.github#97] [Feature requests]
### Is your feature request related to a problem? Please describe.

If a developer begins their project as a plugin or datapack but later expands to support modding platforms and client-side features, then their project is unfortunately stuck as being a plugin/datapack which leaves them without access to features granted to mods such as the "Client side" and "Server side" flags which control how your project is displayed in search results.

### Describe the solution you'd like

Since project goals and plans can change over time, developers should be able to recategorize their project under a different project type so as to ensure they are not locked out of features that may later become useful.

### Describe alternatives you've considered

I could delete and create a new project but that would lose all of the followers and downloads. I could also keep the original project and just maintain two like on C\*rseF\*rge, but this would seem highly redundant and cumbersome.

**Comment by triphora** [comment]
Implemented!

**Reply by qixils** [reply]
Sorry to necro this but I believe there may be some issues with this? My goal is to mark [my mod](https://modrinth.com/plugin/crowdcontrol), which started as a plugin, as functional on both client and server side (currently in search results it shows as just "Server" and is excluded from searches for Client mods) as it contains optional client functionality. The UI options to do so are still missing from the settings page, and manually sending a PATCH to the v2 project API returns with a literally random error about, from what I can tell, the new fields from the v3 API/database.
```sh
curl 'https://api.modrinth.com/v2/project/6XhH9LqD' -X 'PATCH' --data-raw '{"client_side":"optional","server_side":"required"}'
# produces one of the following at random...
{"error":"invalid_input","description":"Invalid Input: Loader field 'singleplayer' does not exist for any loaders supplied."}
{"error":"invalid_input","description":"Invalid Input: Loader field 'client_and_server' does not exist for any loaders supplied."}
{"error":"invalid_input","description":"Invalid Input: Loader field 'client_only' does not exist for any loaders supplied."}
{"error":"invalid_input","description":"Invalid Input: Loader field 'server_only' does not exist for any loaders supplied."}
```
I tried a couple different things with the loader field and even trying to send requests to the v3 API but have been unable to make any changes.
